tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/CredentialTests.cs

```diff
@@ -0,0 +1,1164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.ManagedIdentity;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.TelemetryCore.Internal.Events;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Microsoft.Identity.Test.Common.Core.Helpers.ManagedIdentityTestUtil;
+using Microsoft.Identity.Client.Internal;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+
+namespace Microsoft.Identity.Test.Unit.ManagedIdentityTests
+{
+    /// <summary>
+    /// Unit tests for the Credential component in the Managed Identity scenario.
+    /// </summary>
+    [TestClass]
+    public class CredentialTests : TestBase
+    {
+        internal const string CredentialEndpoint = "http://169.254.169.254/metadata/identity/credential";
+        internal const string MtlsEndpoint = "https://centraluseuap.mtlsauth.microsoft.com/" +
+            "72f988bf-86f1-41af-91ab-2d7cd011db47/oauth2/v2.0/token";
+        internal const string Resource = "https://management.azure.com";
+
+        /// <summary>
+        /// Tests the happy path for acquiring credentials with various CryptoKeyTypes.
+        /// </summary>
+        /// <param name="cryptoKeyType">The type of cryptographic key used.</param>
+        [DataTestMethod]
+        [DataRow(CryptoKeyType.Machine)]
+        [DataRow(CryptoKeyType.User)]
+        [DataRow(CryptoKeyType.InMemory)]
+        [DataRow(CryptoKeyType.Ephemeral)]
+        [DataRow(CryptoKeyType.KeyGuard)]
+        public async Task CredentialHappyPathAsync(int keyType)
+        {
+            CryptoKeyType cryptoKeyType = (CryptoKeyType)keyType;
+
+            // Arrange
+            using (MockHttpAndServiceBundle harness = CreateTestHarness())
+            using (var httpManager = new MockHttpManager(isManagedIdentity: true))
+            {
+                ManagedIdentityApplicationBuilder miBuilder = ManagedIdentityApplicationBuilder
+                    .Create(ManagedIdentityId.SystemAssigned)
+                    .WithHttpManager(httpManager);
+
+                KeyMaterialManagerMock keyManagerMock = new(CertHelper.GetOrCreateTestCert(), cryptoKeyType);
```

Comment:
There is quite a bit of logic in `KeyMaterialManager`. Is it possible to only mock the KeyGuardManager instead? This way you'd test more.